### PR TITLE
PART 2:`use-github-storage` option validation

### PR DESCRIPTION
**Pull request by begonaguereca**
This pull request ensures proper validation of the `--use-github-storage` option.

Closes: https://github.ghe.com/github/octoshift/issues/9326

- [ ] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgs.cs [hunk 1]
```diff
@@ -159,6 +160,10 @@ private void ValidateUploadOptions()
         {
             throw new OctoshiftCliException("The AWS S3 bucket name must be provided with --aws-bucket-name if other AWS S3 upload options are set.");
         }
+        if (UseGithubStorage && AwsBucketName.HasValue())
+        {
+            throw new OctoshiftCliException("The --use-github-storage flag was provided with an AWS S3 Bucket name. Archive cannot be uploaded to both locations.");
+        }
```
Comment (ArinGhazarian) [inline src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:166]: I guess we also need a check for when `--use-github-storage` is provided with `AzureStorageConnectionString`

File: src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs [hunk 2]
```diff
@@ -61,6 +62,15 @@ public override void Validate(OctoLogger log)
...
                 }
+                if (UseGithubStorage)
+                {
+                    throw new OctoshiftCliException("--ghes-api-url must be specified when --use-github-storage is specified.");
+                }
+            }
+
+            if (AwsBucketName.HasValue() && UseGithubStorage)
+            {
+                throw new OctoshiftCliException("The --use-github-storage flag was provided with an AWS S3 Bucket name. Archive cannot be uploaded to both locations.");
```
Comment (ArinGhazarian) [inline src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:73]: I guess we also need a check for when `--use-github-storage` is provided with `AzureStorageConnectionString`

File: src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs [hunk 3]
```diff
@@ -70,6 +70,25 @@ public void It_Throws_When_Aws_Bucket_Name_Not_Provided_But_Aws_Access_Key_Provi
...
 
+        [Fact]
+        public void It_Throws_When_Aws_Bucket_Name_Provided_With_UseGithubStorage_Option()
+        {
+            var args = new MigrateRepoCommandArgs
+            {
+                ArchivePath = ARCHIVE_PATH,
+                GithubOrg = GITHUB_ORG,
+                GithubRepo = GITHUB_REPO,
+                AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING,
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs:81]: Azure storage connection string shouldn't be provided along with aws bucket name. We probably need and extra validation for when both aws and azure options are provided together but probably it'd be a scope creep for this PR. I guess we need to create a ticket to revise our validations and add missing ones.

**Review by ArinGhazarian — COMMENTED** [review]
Mostly looks good, left a couple of comments.

File: src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgs.cs (hunk 1 above)
Comment (begonaguereca) [reply src/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgs.cs:166]: Updated and addressed the comments with [this]([420db29](https://github.com/github/gh-gei/pull/1270/commits/420db29ceb3debcd1d175f0623d6c1a214961d0a)) commit 👍🏽

**Review by ArinGhazarian — APPROVED** [review]
